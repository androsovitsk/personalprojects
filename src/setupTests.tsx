import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import fetchMock from 'fetch-mock'
import TranslationProvider from './ui/components/TranslationProvider/TranslationProvider'
import LanguageTypes from './ui/components/TranslationProvider/types/LanguageTypes'
import resourceBundles from './locale'
import { createTheme, ThemeProvider } from '@mui/material'

beforeEach(() => {
  fetchMock.reset()
  jest.clearAllMocks()
})

const Wrapper: React.FC<React.PropsWithChildren> = ({ children }) => {
  return (
    <ThemeProvider theme={createTheme({})}>
      <TranslationProvider
        defaultLanguage={LanguageTypes.ENGLISH}
        resourceBundles={resourceBundles}
      >
        {children}
      </TranslationProvider>
    </ThemeProvider>
  )
}

export const renderWithProviders = (element: React.ReactElement) =>
  render(element, {
    wrapper: Wrapper
  })
