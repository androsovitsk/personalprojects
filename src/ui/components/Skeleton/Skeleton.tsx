import React from 'react'
import MUISkeleton from '@mui/material/Skeleton'
import { SkeletonProps } from '@mui/material'

interface ISkeletonProps extends SkeletonProps {
  noMargin?: boolean
}

const Skeleton: React.FC<ISkeletonProps> = ({
  noMargin = false,
  children,
  ...props
}) => {
  return (
    <MUISkeleton {...(noMargin && { sx: { marginBottom: 0 } })} {...props}>
      {children}
    </MUISkeleton>
  )
}

Skeleton.displayName = 'Skeleton'

export default Skeleton
