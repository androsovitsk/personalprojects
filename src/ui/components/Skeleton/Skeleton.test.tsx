import { renderWithProviders } from '../../../setupTests'
import Skeleton from './Skeleton'

describe('Skeleton', () => {
  it('should match the snapshot', () => {
    const view = renderWithProviders(<Skeleton />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot when no margin is applied', () => {
    const view = renderWithProviders(<Skeleton noMargin />)

    expect(view.asFragment()).toMatchSnapshot()
  })
})
