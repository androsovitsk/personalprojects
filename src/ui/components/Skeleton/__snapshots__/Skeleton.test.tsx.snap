// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should match the snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
  />
</DocumentFragment>
`;

exports[`Skeleton should match the snapshot when no margin is applied 1`] = `
<DocumentFragment>
  <span
    class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-uxfc48-MuiSkeleton-root"
  />
</DocumentFragment>
`;
